edge-modules/load-gen/Program.cs

```diff
@@ -34,32 +34,32 @@ static async Task Main()
 
                 (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
 
-                // setup the message PeriodicTask
-                using (var messageTask = new PeriodicTask(
-                    () => GenerateMessageAsync(moduleClient, batchId),
-                    Settings.Current.MessageFrequency,
-                    Settings.Current.StartDelay,
-                    Logger,
-                    "Generate message"))
+                Logger.LogInformation("Load gen running.");
+                Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+                await Task.Delay(Settings.Current.TestStartDelay).ConfigureAwait(false);
+
+                DateTime testStartAt = DateTime.UtcNow;
+                while (!cts.IsCancellationRequested && DateTime.UtcNow - testStartAt < Settings.Current.TestDuration)
                 {
-                    Logger.LogInformation("Load gen running.");
-                    await cts.Token.WhenCanceled();
+                    await GenerateMessageAsync(moduleClient, batchId, Settings.Current.TrackingId);
+                    await Task.Delay(Settings.Current.MessageFrequency).ConfigureAwait(false);
```

Comment:
Good question.  In general configureAwait(false) is recommended to use if context switch is not needed.  However it is not used in our edge solution as one of our dependencies, dotnetty requires context synchronization; if configureAwait(false) is used, it will cause unexpected error.  That's why you can't find it used in EdgeAgent/EdgeHub.

I used here as it is applied to Task.Delay; not anything related to SDK.  If you have any concerns, I am ok to remove it.

Reply:
My understanding is that .NET core does not support SynchronizationContext and that is the reason configure await false is not needed. I think I am misunderstanding something because if dotnetty requires SynchronizationContext, then I would think that would mean:
- we would not be able to run on .NET core, since .NET core does not support it
- configure await false might be needed to avoid context switches to synchronous methods, but this error you mention would not allow us to do so

Reply:
We don't use ConfigureAwait anywhere in our code, please remove it from here. It should not be needed.

Reply:
remove COnfigureAwait?